packages/error/src/index.ts

```diff
@@ -1,11 +1,9 @@
-//Note: This class only exists for compatibility with some old Javascript
-//stuff that avoided using Error directly for whatever reason.  Eventually
-//it should be eliminated.
-class ExtendableError extends Error {
+class TruffleError extends Error {
```

Comment:
can we just get rid of this package entirely? the comments that you removed seem to imply that we don't need it any more. we don't really use it that much, do we?

Reply:
I don't think we can remove this because [`@truffle/error` is used](https://github.com/trufflesuite/truffle/blob/develop/packages/core/cli.js#L98) in situations when we know exactly why an error condition exists and should inform the user to take a specific action. For example, if they use `truffle migrate --url xyz --network falafel`, this is a truffle error condition, and we can use `@truffle/error` to let the user know they shouldn't use mutually exclusive options in the `migrate` command. In this situation, showing a stack trace would be confusing.

Reply:
so you are saying you think this package is still useful - I would like to know exactly when to use this type of error in the future. would the `testConnection` method [here](https://github.com/trufflesuite/truffle/blob/develop/packages/provider/index.js#L36-L65) be an appropriate spot to utilize this error type?

Reply:
That does seem like a good place to utilize `@truffle/error` because there's enough information to give the user a corrective action.

I'm not sure about `ProviderError`, and would have to check how it's used elsewhere.

Reply:
I think we should have some kind of standard policy for `TruffleError` - I never use this at all and have no concept of when it is correct to do so other than what you've written here on this PR